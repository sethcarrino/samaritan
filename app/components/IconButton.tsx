import React from 'react';
import { TouchableOpacity, StyleSheet, ViewStyle } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

interface Props {
  icon: string;
  size: number;
  color: string;
  noGutter?: boolean;
  onPress?: () => void;
}

const IconButton: React.FC<Props> = ({
  icon,
  size,
  color,
  noGutter,
  onPress
}) => (
  <TouchableOpacity
    onPress={onPress ? onPress : e => e.preventDefault()}
    style={{ marginHorizontal: noGutter ? 0 : 20 }}
  >
    <Ionicons name={icon} size={size} color={color} />
  </TouchableOpacity>
);

type IconButtonStyleSheet = {
  container: ViewStyle;
};

export default IconButton;
