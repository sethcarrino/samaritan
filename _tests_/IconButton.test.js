import React from 'react';
import { mount, configure } from 'enzyme';
import renderer from 'react-test-renderer';

import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import IconButton from '../app/components/IconButton';

describe('<IconButton />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<IconButton />).toJSON();
    expect(tree.children.length).toBe(1);
  });
  it('renders correctly', () => {
    const tree = renderer.create(<IconButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
