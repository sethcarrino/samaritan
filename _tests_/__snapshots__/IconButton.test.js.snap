// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginHorizontal": 20,
      "opacity": 1,
    }
  }
>
  <Text />
</View>
`;
